lib/bundle/index.js

```diff
@@ -145,7 +145,7 @@ function inventory$Ref ($refParent, $refKey, path, pathFromRoot, indirections, i
     }
   }
 
-  if (options.bundle.generateKey && file !== $refs._root$Ref.path) {
+  if (options.bundle.generateKey && file !== $refs._root$Ref.path && !(path.indexOf($refs._root$Ref.path) === 0 && $ref.$ref.indexOf("#/") === 0)) {
```

Comment:
` !(path.indexOf($refs._root$Ref.path) === 0 && $ref.$ref.indexOf("#/") === 0)`
= this is to avoid calling that generate key fn if a given $ref is local and already in root file

there is no point in doing that since it's already a local $ref, therefore it does not need any hoisting